data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBufferAccumulator.java

```diff
@@ -47,31 +44,23 @@ public class KafkaSourceBufferAccumulator<K, V> {
     private static final Logger LOG = LoggerFactory.getLogger(KafkaSourceBufferAccumulator.class);
     private static final String MESSAGE_KEY = "message";
     private final TopicConfig topicConfig;
-    private final KafkaSourceConfig kafkaSourceConfig;
     private final String schemaType;
-    private PluginMetrics pluginMetrics;
-    private final Counter kafkaConsumerWriteError;
-    private static final String KAFKA_CONSUMER_BUFFER_WRITE_ERROR = "kafkaConsumerBufferWriteError";
     private static final int MAX_FLUSH_RETRIES_ON_IO_EXCEPTION = Integer.MAX_VALUE;
     private static final Duration INITIAL_FLUSH_RETRY_DELAY_ON_IO_EXCEPTION = Duration.ofSeconds(5);
     private final JsonFactory jsonFactory = new JsonFactory();
     private static final ObjectMapper objectMapper = new ObjectMapper();
     private static final Long COMMIT_OFFSET_INTERVAL_MILLI_SEC = 300000L;
 
     public KafkaSourceBufferAccumulator(final TopicConfig topicConfigs,
-                                        final KafkaSourceConfig kafkaSourceConfig,
                                         final String schemaType, PluginMetrics pluginMetric) {
-        this.kafkaSourceConfig = kafkaSourceConfig;
         this.topicConfig = topicConfigs;
         this.schemaType = schemaType;
-        this.pluginMetrics = pluginMetric;
-        this.kafkaConsumerWriteError = pluginMetrics.counter(KAFKA_CONSUMER_BUFFER_WRITE_ERROR);
     }
 
     public Record<Object> getEventRecord(final String line) {
```

Comment:
Is this the right place for this? Passing the schema type to buffer accumulator seems wrong because buffer accumulator should accumulate "events" and the creation of event should happen in the KafkaSourceCustomConsumer which knows what type of schema it is handling.

Reply:
We have the parsing logic for Plaintext/Jason and Avro written in the [/KafkaSourceBufferAccumulator.java] and such parsing method requires the schemType to parse the data.